src/classes/BDI_DataImportService.cls

```diff
@@ -704,20 +704,54 @@ global with sharing class BDI_DataImportService {
     * from the list of DataImport Fields, which have the form DIObject.developerfieldname in the field's help text property.
     * @param dataImportObjectName the bdi object we care about (Contact1, Contact2, Account1, etc)
     * @param targetObjectName the underlying object the bdi object is (ie, Contact, Account, etc)
-    * @param listStrDIFields the DataImport fields to consider
+    * @param dataImportFields the DataImport fields to consider
     * @return Map<String, String> a map of DataImport fields to underlying object fields (eg, Donation_Amount__c => Amount)
     */
     global static Map<String, String> mapFieldsForDIObject(String dataImportObjectName, String targetObjectName, List<String> dataImportFields) {
 
+        BDISettings__c bdiSettings = BDISettings__c.getOrgDefaults();
+
         Map<String, String> dataImportFieldToTargetField = new Map<String, String>();
+        Map<String, String> bdiFIeldMappingMap = new Map<String, String>();
+
+        Boolean useCustomMetadatainsteadOfHelpText = false;
+
+        if (bdiSettings.UseCustomMetadatainsteadOfHelpText__c) {
```

Comment:
This field name could be clearer. Maybe MapWithCustomMetadataTypes__c? I'd like it to say something about what we're doing (mapping) and leaving the reference to help text in the name could get stale over time. This comment also applies to your Boolean variable (useCustomMetadatainsteadOfHelpText -> useCustomMetadataForFieldMapping).

Reply:
Verified changes.